### ISSUE TITLE
CSS Structure

### ISSUE THREAD

**Issue by mttmccb** [tailwindcss/discuss#58]
Just wondering about this part of the docs

"Since Tailwind's utility classes don't rely on `!important` to defeat other styles, it's important that you add your component classes before any utility classes in your CSS."

Might make it difficult to use tailwind as I want to encapsulate my component css with the component, is there a recommended approach to handle this? It would likely be loaded after the tailwind css, just wondering if there is maybe an `use important` flag or if that would make sense so it would work as expect in this scenario?

**Comment by adamwathan** [comment]
> ...as I want to encapsulate my component css with the component

Do you mean component like React/Vue component in this case? If so yeah, I'm not sure where those styles get added relative to all of your other styles. It seems like this should be researchable though, given how important declaration order is in CSS I doubt it's something that hasn't been talked about.

That said, in the next release you will be able to make all of Tailwind's utilities `!important` by adding an option to your config file 👍 

https://github.com/tailwindcss/tailwindcss/pull/186

**Comment by mttmccb** [comment]
Yeah it's a bit like React/Vue, however it's an external CSS file (slightly different when webpack gets involved), it seems to be injected when that component is used (not the only way it can be done).

It sounds like making them `!important` would solve my issue if I don't want to separate my component css, I'll keep an eye on that issue.
